### PR TITLE
EQL: Hook engine to Elasticsearch

### PR DESCRIPTION
**Pull request by costin**
Add query execution and return actual results returned from
Elasticsearch inside the tests

### CODE REVIEW

**Comment by jrodewig** [comment]
It looks like elasticsearch-ci/oss-distro-docs is failing because there isn't an explicit `event_type_field` for the snippet here:
https://www.elastic.co/guide/en/elasticsearch/reference/master/eql-search.html

I can easily add one to that snippet if `event.category` will no longer be the default value. Excited to see this!

File: x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/plugin/TransportEqlSearchAction.java [hunk 1]
```diff
@@ -68,23 +66,17 @@ public static void operation(PlanExecutor planExecutor, EqlSearchRequest request
...
-        listener.onResponse(createResponse(null));
+        Configuration cfg = new Configuration(request.indices(), zoneId, username, clusterName, filter, timeout, request.fetchSize(),
+                includeFrozen, clientId);
+        planExecutor.eql(cfg, request.rule(), params, wrap(r -> listener.onResponse(createResponse(r)), listener::onFailure));
     }
 
     static EqlSearchResponse createResponse(Results results) {
-        // Stubbed search response
-        // TODO: implement actual search response processing once the parser/executor is in place
         // Updated for stubbed response to: process where serial_event_id = 1
```
Comment (aleksmaus) [inline x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/plugin/TransportEqlSearchAction.java]: nit: can remove the comment above

**Review by aleksmaus — APPROVED** [review]
Very nice to see it wired 👍 
Thank you!

File: docs/reference/eql/search.asciidoc [hunk 2]
```diff
@@ -42,5 +42,5 @@ GET sec_logs/_eql/search
 // TEST[continued]
```
Comment (jrodewig) [inline docs/reference/eql/search.asciidoc]: Github wont' let me comment on the snippet above, but it should be updated to something like:

```
[source,console]
----
GET sec_logs/_eql/search
{
  "event_type_field": "event.category",
 "rule": """
    process where process.name == "cmd.exe"
  """
----
}
```
I would leave `event.category` as the `event_type_field` as it conforms to ECS.

File: docs/reference/eql/search.asciidoc [hunk 3]
```diff
@@ -17,18 +17,18 @@ The following <<docs-bulk,bulk API>> request adds some example log data to the
...
+{ "@timestamp": "2020-12-07T11:06:07.000Z", "agent": { "id": "8a4f500d" }, "event_type": "process" , "process": { "name": "cmd.exe", "path": "C:\\Windows\\System32\\cmd.exe" } }
 {"index":{"_index" : "sec_logs"}}
-{ "@timestamp": "2020-12-07T11:07:08.000Z", "agent": { "id": "8a4f500d" }, "event": { "category": "image_load" }, "file": { "name": "cmd.exe", "path": "C:\\Windows\\System32\\cmd.exe" }, "process": { "name": "cmd.exe", "path": "C:\\Windows\\System32\\cmd.exe" } }
+{ "@timestamp": "2020-12-07T11:07:08.000Z", "agent": { "id": "8a4f500d" }, "event_type": "image_load" , "file": { "name": "cmd.exe", "path": "C:\\Windows\\System32\\cmd.exe" }, "process": { "name": "cmd.exe", "path": "C:\\Windows\\System32\\cmd.exe" } }
 {"index":{"_index" : "sec_logs"}}
-{ "@timestamp": "2020-12-07T11:07:09.000Z", "agent": { "id": "8a4f500d" }, "event": { "category": "process" }, "process": { "name": "regsvr32.exe", "path": "C:\\Windows\\System32\\regsvr32.exe" } }
+{ "@timestamp": "2020-12-07T11:07:09.000Z", "agent": { "id": "8a4f500d" }, "event_type": "process" , "process": { "name": "regsvr32.exe", "path": "C:\\Windows\\System32\\regsvr32.exe" } }
```
Comment (jrodewig) [inline docs/reference/eql/search.asciidoc]: I would leave this as-is and just update the EQL search snippet below.

File: docs/reference/eql/search.asciidoc (hunk 2 above)
Comment (costin) [reply docs/reference/eql/search.asciidoc]: Thanks @jrodewig - I'll follow-up with a different PR to fix the default `event_type_field` to `event.category` not just in docs, but in the tests as well.

Comment (jrodewig) [reply docs/reference/eql/search.asciidoc]: That works for me. Thanks!

Comment (costin) [reply docs/reference/eql/search.asciidoc]: Looked at the code and figured that the `event_type_field` is actually implemented and your suggestions worked flawlessly.
Thanks!

**Review by jrodewig — APPROVED** [review]
Docs LGTM. Thanks!

File: x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/session/Results.java [hunk 4]
```diff
@@ -8,28 +8,76 @@
...
+        COUNT;
+    }
+    
+    public static final Results EMPTY = new Results(new TotalHits(0, Relation.EQUAL_TO), TimeValue.MINUS_ONE, false, emptyList());
 
     private final TotalHits totalHits;
-    private final List<Object> results;
+    private final List<?> results;
+    private final boolean timedOut;
+    private final TimeValue time;
```
Comment (astefan) [inline x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/session/Results.java]: Is `time` the amount of time it took to return the results (judging by the `execTime()` method) ? If so, wouldn't be more descriptive to have this named `took` or `duration` or `execTime`?

**Review by astefan — APPROVED** [review]
LGTM. Left only one minor comment.

**Pull request merged**